packages/ra-ui-materialui/src/types.ts

```diff
@@ -73,7 +77,11 @@ export interface ShowProps<RecordType extends RaRecord = any> {
     id?: Identifier;
     queryOptions?: UseQueryOptions<RecordType> & { meta?: any };
     resource?: string;
-    title?: string | ReactElement;
+    title?:
+        | string
+        | ReactElement
+        | false
+        | ((arg0: RecordType | undefined) => string);
```

Comment:
<!-- Polish comments are like a suggestion, where there is nothing necessarily wrong with the relevant content, there's just some ways to immediately improve the quality.  -->
**polish:** ‏

```suggestion
        | ((record: RecordType | undefined) => string);
```